src/mapgen/mapgen_v6.cpp

```diff
@@ -807,51 +806,44 @@ void MapgenV6::flowMud(s16 &mudflow_minpos, s16 &mudflow_maxpos)
 
 					VoxelArea::add_y(em, i, -1);
 				}
-
-				// Stop if out of area
-				//if(vmanip.m_area.contains(i) == false)
+				// No mud found in mapchunk column, process the next column
 				if (y < node_min.Y)
 					break;
 
 				if (n->getContent() == c_dirt ||
 						n->getContent() == c_dirt_with_grass) {
-					// Make it exactly mud
+					// Convert dirt_with_grass to dirt
 					n->setContent(c_dirt);
-
-					// Don't flow it if the stuff under it is not mud
+					// Don't flow it if the stuff under it is not mud,
+					// to leave at least 1 node of mud.
 					{
 						u32 i2 = i;
 						VoxelArea::add_y(em, i2, -1);
-						// Cancel if out of area
-						if (!vm->m_area.contains(i2))
-							continue;
```

Comment:
Unnecessary check because:
At this point in code, y is no lower than node_min.Y (mapchunk base), voxelmanip base is 16 nodes below node_min.Y, so y - 1 is obviously inside area.